benchmark/result/result.py

```diff
@@ -0,0 +1,48 @@
+"""Basic result classes for benchmarking."""
+
+from typing import Any
+from uuid import UUID
+
+from attrs import define, field
+from attrs.validators import instance_of
+from pandas import DataFrame
+
+
+@define(frozen=True)
+class Result:
+    """A single result of the benchmarking."""
+
+    name: str = field(validator=instance_of(str))
+    """The name of the benchmarking result."""
+
+    identifier: UUID = field(validator=instance_of(UUID))
+    """The unique identifier of the benchmark running which can be set
+    to compare different executions of the same benchmark setting."""
+
+    metadata: dict[str, str] = field()
+    """Metadata about the benchmarking result."""
+
+    benchmark_result: DataFrame = field()
+    """The result of the benchmarked callable."""
+
+    execution_time_ns: int = field(validator=instance_of(int))
+    """The execution time of the benchmark in nanoseconds."""
+
+    @metadata.validator
+    def _convert_metadata_to_string(self, _: Any, metadata: dict[Any, Any]) -> None:
```

Comment:
This function is currently not used at all (resp. only for validation without actually doing any validation), why is it then already included?

Reply:
It has been restructured to be a converter, and I thought I had already pushed that change. However, it should definitely have been removed by now.